### DISCUSSION TITLE
How can I get error messages set in HTTP responses created from BadRequestException?

### DISCUSSION THREAD

**Discussion by davechallis** [quarkusio/quarkus#36243] [Q&A] [answered]
I'm just starting out with Quarkus, and trying a simple resource:

```java
@Path("/")
public class FooResource {
    @Produces(MediaType.APPLICATION_JSON)
    @GET
    @Path("/items")
    public List<Integer> items(@RestQuery final String filter) {
        if (filter == null) {
            throw new BadRequestException("missing 'filter' parameter");
        }
        return Collections.emptyList();
    }
}
```
If I test this locally using curl, I get the correct 400 status code, but no message body:
```shell
$ curl -i localhost:80/items
HTTP/1.1 400 Bad Request
content-length: 0

```
Is there some I need to change to get the Exception's message set as part of the response body?

[a comment by an automated account is not shown]

**Comment by sberyozkin** [comment]
CC @geoand

**Comment by luca-bassoricci** [comment]
The fastest way is to add `jakarta.validation.constraints.@NotNull` to `filter` parameter, remove your test and add `io.quarkus:quarkus-hibernate-validator` dependency to your pom.
This should return a response of type `io.quarkus.hibernate.validator.runtime.jaxrs.ViolationReport` when filter is `null`.<br/>

**Reply by davechallis** [reply, marked as answer]
Thanks, that also works. I was mostly wondering about the general approach though, in my application I'm passing one of the parameters to another service, so don't know whether it's an invalid parameter until part way through the method execution.

I solved it in the end by creating a mapper for the exception instead:
```java
    @ServerExceptionMapper
    public RestResponse<String> mapException(BadRequestException e) {
        return RestResponse.status(RestResponse.Status.BAD_REQUEST, e.getMessage());
    }
```

**Reply by geoand** [reply]
The general idea is that Quarkus itself most of the time does not provide a body for the error messages as that could potentially reveal internal information

**Reply by omasseau** [reply]
> The general idea is that Quarkus itself most of the time does not provide a body for the error messages as that could potentially reveal internal information

There should be a way to get the error message in the body or in the logs when in dev or test mode.
